flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1528,6 +1532,51 @@ IntrinsicLibrary::genPresent(mlir::Type,
                                           fir::getBase(args[0]));
 }
 
+// SCAN 
+fir::ExtendedValue
+IntrinsicLibrary::genScan(mlir::Type resultType,
+                          llvm::ArrayRef<fir::ExtendedValue> args) {
+
+  assert(args.size() >= 2);
```

Comment:
I think you can and should assert `== 4` here, (You are using `args[3]` below and args should always be of sized 4 for scan, even if two of the args might be absent).